### DISCUSSION TITLE
use version property in vue which defined from package.json file

### DISCUSSION THREAD

**Discussion by honwhy** [wxt-dev/wxt#909] [Ideas]
since wxt dev/build  can use version from package.json define property version,  can we use this version property in such as extension options page,
suppose something like this
```
<template>
<div>my project {{import.meta.env.VERSION}}</div>
</template>
```

check this [#version](https://wxt.dev/guide/key-concepts/manifest.html#version)
and this [vite meta env](https://vitejs.dev/guide/env-and-mode)

**Comment by honwhy** [comment]
I managed to figure out one solution,
```
// wxt.config.ts
import * as pack from './package.json'
export default defineConfig({
...
  vite: () => ({
    define: {
      __APP_VERSION__: JSON.stringify(pack.version),
    },
  })
})
```
use in <setup>
```
<script setup lang="ts">
const appVersion = ref(__APP_VERSION__)
</script>
```
maybe define global variable for tsconfig.json
```
declare global {
  const __APP_VERSION__: string
}
```

**Reply by aklinker1** [reply]
Yup! Alternatively: `browser.runtime.getManifest().version`

**Reply by honwhy** [reply]
good
